### PR TITLE
Optionally pad OpenPGP EdDSA signature parameters

### PR DESCRIPTION
**Pull request by lukpueh**
Fixes: #334

### Description of the changes being introduced by the pull request:
Left-zero-pad 'r' and 's' values that are shorter than required by RFC 8032 (5.1.6.), to make up for omitted leading zeros in RFC 4880 (3.2.) MPIs. This is especially important for 's', which is little-endian.

Note that GnuPG seems to correctly parse EdDSA signatures with MPIs that omit leading zeros prior to validation. But in order to validate these signatures outside the context of RFC 4880, e.g. with pyca/cryptography's OpenSSL backend, we need to guarantee the correct length of the parsed signatures ourselves.

This PR also adds a test using a special-crafted short signature.  

See commit messages and #334 for more details.

Kudos to @jku and @SantiagoTorres for helping to troubleshoot.

### Please verify and check that the pull request fulfils the following requirements:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: securesystemslib/gpg/eddsa.py [hunk 1]
```diff
@@ -140,6 +142,12 @@ def get_signature_params(data):
   ptr += 2
   s = data[ptr:ptr + s_length]
 
+  # Left-zero-pad 'r' and 's' values that are shorter than required by RFC 8032
+  # (5.1.6.), to make up for omitted leading zeros in RFC 4880 (3.2.) MPIs.
+  # This is especially important for 's', which is little-endian.
+  r = r.rjust(int(ED25519_SIG_LENGTH / 2), b"\x00")
+  s = s.rjust(int(ED25519_SIG_LENGTH / 2), b"\x00")
```
Comment (jku) [inline securesystemslib/gpg/eddsa.py:149]: not a big deal but I think this should work in all pythons:
```suggestion
  r = r.rjust(ED25519_SIG_LENGTH // 2, b"\x00")
  s = s.rjust(ED25519_SIG_LENGTH // 2, b"\x00")
```

File: tests/test_gpg.py [hunk 2]
```diff
@@ -778,5 +779,24 @@ def test_gpg_sign_and_verify_object_with_specific_key(self):
...
+    test_data = b"hello"
+    signature_path = os.path.join(self.gnupg_home, "short.sig")
+
+    # Read special-crafted raw gpg signature that is one byte too short
+    with open(signature_path, "rb") as f:
+      signature_data = f.read()
+
+    # Check that the signature is padded upon parsing
+    signature = parse_signature_packet(signature_data)
+    self.assertTrue(len(signature["signature"]) == (ED25519_SIG_LENGTH * 2))
```
Comment (jku) [inline tests/test_gpg.py:794]: So this is double length here because signature is already a hex string at this point?

**Review by jku — APPROVED** [review]
LGTM

File: securesystemslib/gpg/eddsa.py (hunk 1 above)
Comment (lukpueh) [reply securesystemslib/gpg/eddsa.py:149]: Nice use of the floor division operator. Will update.

File: tests/test_gpg.py (hunk 2 above)
Comment (lukpueh) [reply tests/test_gpg.py:794]: Exactly. I'll add a comment.

**Comment by lukpueh** [comment]
Addressed your comments and force-pushed. Thanks for the review, @jku!

**Pull request merged**